views.py

```diff
@@ -0,0 +1,79 @@
+from django.shortcuts import render, render_to_response, get_object_or_404, redirect
+from django.http import HttpResponseRedirect, HttpResponse, JsonResponse
+from django.template.context_processors import csrf
+from django.template import RequestContext
+import json
+import requests
+import urllib.parse
+from collections import defaultdict
+import datetime
+import dateutil.parser
+import json
+
+def home(request):
+	if request.method == 'POST' and request.is_ajax():
```

Comment:
You'd want a separate URL for this (with GET, since it does not change any data); and then you have provided a REST API as well.

Also, if it does not work without JS, it should at least tell that to the user.

Reply:
I'm now showing this message that graphs cannot be created if javascript is disabled in user's browser.

And, I'm sorry but I guess I didn't get the first part of your comment completely. Do you want me to make two separate URLs and views for GET and POST request? Won't we have
 if request.method == 'POST' and request.is_ajax(): and if request.method == 'GET' there?

Reply:
I'm now showing this message that "graphs cannot be created" if javascript is disabled in user's browser!

And, Sorry but I guess I didn't get the first part of your comment completely. Do you want me to make two separate URLs and views for GET and POST request? Won't we be writing 
if request.method == 'POST' and request.is_ajax(): there too?

Reply:
There isn't really any reason to use POST here; you are not updating/creating a resource. Responses to POST requests usually do not get cached, and there is no reason to disallow the browser from caching this data.

The nice approach would be to have two routes, `/` which returns a HTML reponse (the one you return here on the GET / non-AJAX branch) and `/api/user/{username}` (or something like that) which returns a JSON response (the one you return here on the POST AJAX branch). You don't need to care about the request method or whether it is AJAX.

That way, besides the code structure being a little easier to parse, you have also provided a public API for free (without making clients jump through hooks to pretend they are making an AJAX call).

If you did support non-JS clients, that kind of thing would be more complicated, as you'd have to support users submitting the form in the regular way, and then compose the response on the server side. But the Google Charts API probably doesn't work very well that way (since it would see all requests coming from your server IP and you'd hit quotas quickly) so no point in worrying about that.